services/grafana/client.go

```diff
@@ -300,36 +311,89 @@ func (c *Client) testDeleteUser(ctx context.Context, userID int, authHeaders htt
 	return c.do(ctx, "DELETE", "/api/admin/users/"+strconv.Itoa(userID), "", authHeaders, nil, nil)
 }
 
-func (c *Client) testCreateAPIKey(ctx context.Context, name string, role role, authHeaders http.Header) (int, string, error) {
+// CreateAdminAPIKey creates API key with Admin role and provided name.
+func (c *Client) CreateAdminAPIKey(ctx context.Context, name string) (int64, string, error) {
+	authHeaders, err := c.authHeadersFromContext(ctx)
+	if err != nil {
+		return 0, "", err
+	}
+	return c.createAPIKey(ctx, name, admin, authHeaders)
+}
+
+// DeleteAPIKeysWithPrefix deletes all API keys with provided prefix. If there is no api key with provided prefix just ignores it.
+func (c *Client) DeleteAPIKeysWithPrefix(ctx context.Context, prefix string) error {
+	authHeaders, err := c.authHeadersFromContext(ctx)
+	if err != nil {
+		return err
+	}
+	var keys []apiKey
+	if err := c.do(ctx, "GET", "/api/auth/keys", "", authHeaders, nil, &keys); err != nil {
+		return err
+	}
+
+	for _, k := range keys {
+		if strings.HasPrefix(k.Name, prefix) {
+			err := c.deleteAPIKey(ctx, k.Id, authHeaders)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *Client) authHeadersFromContext(ctx context.Context) (http.Header, error) {
+	headers, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, fmt.Errorf("cannot get headers from metadata %v", headers)
```

Comment:
`headers` will be nil because `ok == false`

Reply:
No, it won't be nil :) It will be zero value.
I'll update error message